README.md

```diff
@@ -20,92 +20,130 @@ nvm install stable
+### Run the backend and frontend using `docker`
 
-At that point, you can already open your browser at [http://localhost](http://localhost/) and see that telemetry is waiting for data.
+If you'd like to get things runing manually using Docker, you can do the following. This assumes that you've built the images as per the above, and have two images named `substrate-telemetry-backend` and `substrate-telemetry-frontend`.
 
-Let's bring some data in with  a node:
+1. Create a new shared network so that the various containers can communicate with eachother:
 
-```sh
-docker run --rm -i --name substrate --link backend -p 9944:9944 \
-  chevdor/substrate substrate --dev --telemetry-url 'ws://backend:8000/submit 0'
-```
+   ```
+   docker network create telemetry
+   ```
 
-You should now see your node showing up in your local [telemetry frontend](http://localhost/):
-![image](doc/screenshot01.png)
+2. Start up the backend core process. We expose port 8000 so that a UI running in a host browser can connect to the `/feed` endpoint.
 
-### Run via docker-compose
+   ```
+   docker run --rm -it --network=telemetry \
+       --name backend-core \
+       -p 8000:8000 \
+       substrate-telemetry-backend \
+       telemetry_core -l 0.0.0.0:8000
+   ```
 
-To run via docker make sure that you have Docker Desktop.
-If you don't you can download for you OS here [Docker Desktop](https://www.docker.com/products/docker-desktop)
+3. In another terminal, start up the backend shard process. We tell it where it can reach the core to send messages (possible because it has been started on the same network), and we listen on and expose port 8001 so that nodes running in the host can connect and send telemetry to it.
+
+   ```
+   docker run --rm -it --network=telemetry \
+       --name backend-shard \
+       -p 8001:8001 \
+       substrate-telemetry-backend \
+       telemetry_shard -l 0.0.0.0:8001 -c http://backend-core:8000/shard_submit
+   ```
+
+4. In another terminal, start up the frontend server. We pass a `SUBSTRATE_TELEMETRY_URL` env var to tell the UI how to connect to the core process to receive telemetry. This is relative to the host machine, since that is where the browser and UI will be running.
+
+   ```
+   docker run --rm -it --network=telemetry \
+       --name frontend \
+       -p 3000:80 \
+       -e SUBSTRATE_TELEMETRY_URL=ws://localhost:8000/feed \
+       substrate-telemetry-frontend
+   ```
+
+   **NOTE:** Here we used `SUBSTRATE_TELEMETRY_URL=ws://localhost:8000/feed`. This will work if you test with everything running locally on your machine but NOT if your backend runs on a remote server. Keep in mind that the frontend docker image is serving a static site running your browser. The `SUBSTRATE_TELEMETRY_URL` is the WebSocket url that your browser will use to reach the backend. Say your backend runs on a remore server at `192.168.0.100`, you will need to set the IP/url accordingly in `SUBSTRATE_TELEMETRY_URL`.
```

Comment:
note sure if this is correct because it's a little weird to run `remote server` at a local IP address but I guess it just an example

```suggestion
   **NOTE:** Here we used `SUBSTRATE_TELEMETRY_URL=ws://localhost:8000/feed`. This will work if you test with everything running locally on your machine but NOT if your backend runs on a remote server. Keep in mind that the frontend docker image is serving a static site running your browser. The `SUBSTRATE_TELEMETRY_URL` is the WebSocket url that your browser will use to reach the backend. Say your backend runs on a remote server at `192.168.0.100`, you will need to set the IP/url accordingly in `SUBSTRATE_TELEMETRY_URL`.
```

Reply:
Yeah, `192.168.0.100` isn't the best example address really 😄